src/Library/types.ts

```diff
@@ -0,0 +1,85 @@
+export type Author = {
+    readonly name: string;
+    readonly url: string;
+    readonly position: string;
+};
+
+enum DefaultCategories {
+    All = 'All',
+    Popular = 'Popular'
+}
+
+enum PartnersCategories {
+    Reastream = 'Reastream'
+}
+
+export type LibraryCategories = DefaultCategories | PartnersCategories;
+
+export enum LibraryDirections {
+    All = 'All',
+    Graphics = 'Graphics',
+    Dev = 'Dev',
+    Marketing = 'Marketing',
+    Management = 'Management',
+    Advertising = 'Advertising'
+}
+
+export enum LibraryFormats {
+    All = 'All',
+    Conference = 'Conference',
+    Video = 'Video',
+    Tutorial = 'Tutorial',
+    Interview = 'Interview'
+}
+
+export enum LibraryLevels {
+    All = 'All',
+    Beginner = 'Beginner',
+    Specialist = 'Specialist'
+}
+
+enum VideoSize {
+    Default = 'Default',
+    Featured = 'Featured'
+}
+
+export interface Video {
+    readonly id: number;
+    readonly title: string;
+    readonly description: string;
+    readonly mainCategory?: LibraryCategories;
+    readonly categories: LibraryCategories[];
+    readonly publishDate?: string;
+    readonly authors: Author[];
+    readonly preview: string;
+    readonly source?: string;
+    readonly isFavorite: boolean;
+    readonly isPrivate: boolean;
+    readonly videoSize: VideoSize;
+}
+
+export type VideoСompilation = {
+    readonly title: string;
+    readonly link: string;
+    readonly list: Video[] | null;
```

Comment:
Seems it doesn't make sense to expect `null` value.
If u r going to iterate the array, an empty one works fine, and there's no need to check `null` values.

Reply:
Good point, thanks.